Classes/common/CDTIncrementalStore.m

```diff
@@ -2208,6 +2221,75 @@ - (id)executeSaveRequest:(NSSaveChangesRequest *)saveRequest
     return @[];
 }
 
+- (id)executeBatchUpdateRequest:(NSBatchUpdateRequest *)updateRequest
+                    withContext:(NSManagedObjectContext *)context
+                          error:(NSError **)error
+{
+    NSError *err = nil;
+
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    fetchRequest.entity = updateRequest.entity;
+    fetchRequest.predicate = updateRequest.predicate;
+
+    NSDictionary *query = [self queryForFetchRequest:fetchRequest error:&err];
+
+    if (!query) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    err = nil;
+    CDTQueryResult *hits = [self.indexManager queryWithDictionary:query options:nil error:&err];
+    // hits == nil is valid, get rid of this once tested
+    if (!hits && err) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    // Note: This dictionary is *not* the same dictionary specified in the original request.
+    // The keys have all been transformed from attribute names into NSAttributeDescriptions
+    NSDictionary *updates = updateRequest.propertiesToUpdate;
+
+    NSEntityDescription *entity = [updateRequest entity];
+
+    NSMutableArray *results = [NSMutableArray array];
+    for (CDTDocumentRevision *rev in hits) {
+        NSManagedObjectID *moid = [self newObjectIDForEntity:entity referenceObject:rev.docId];
+        NSManagedObject *mo = [context objectWithID:moid];
+
+        for (NSAttributeDescription *attr in [updates allKeys]) {
+            id newValue = [updates[attr] constantValue];
+            [mo setValue:newValue forKey:attr.name];
+        }
+
+        if (![self updateManagedObject:mo error:&err]) {
+            if (error) *error = err;
+            return nil;
```

Comment:
I'm concerned that if this fails after some have been successfully updated the user will not know what got updated.
I have the same problem with all the save request, so we should discuss this, or find out what Apple thinks about "midway failures"